main.go

```diff
@@ -89,40 +92,109 @@ func isDirEmpty(dirPth string) (bool, error) {
 	} else if empty {
-		return false, nil
+		buildDirAvailable = false
+		log.Errorf("empty")
+	} else {
+		log.Successf("found")
 	}
+
 	// ---
 
 	// read cache indicator file (Carthage/Cachefile)
+	fmt.Print("- Cachfile: ")
+	var cacheFileContent string
+
 	cacheFilePth := filepath.Join(carthageDir, cacheFileName)
-	cacheFileContent, err := contentOfFile(cacheFilePth)
-	if err != nil {
+	if exist, err := pathutil.IsPathExists(cacheFilePth); err != nil {
 		return false, err
+	} else if !exist {
+		cacheFileAvailable = false
+		log.Errorf("not available yet")
+	} else {
+		log.Successf("found")
+		log.Debugf(cacheFileName + " exists in " + cacheFilePth + "\n")
+
+		cacheFileContent, err = contentOfFile(cacheFilePth)
+		if err != nil {
+			return false, err
+		} else if cacheFileContent == "" {
+			log.Errorf("Cachfile is empty")
+		} else {
+			log.Debugf(cacheFileName + " content: " + cacheFileContent + "\n")
+		}
 	}
-	if cacheFileContent == "" {
-		return false, nil
-	}
+
 	// --
 
 	// read Cartfile.resolved
+	fmt.Print("- Cartfile.resolved: ")
+	var resolvedFileContent string
+
 	resolvedFilePath := filepath.Join(srcDir, resolvedFileName)
-	resolvedFileContent, err := contentOfFile(resolvedFilePath)
-	if err != nil {
+	if exist, err := pathutil.IsPathExists(resolvedFileName); err != nil {
 		return false, err
+	} else if !exist {
+		log.Errorf("not found")
+		resolvedFileAvailable = false
+	} else {
+		log.Successf("found")
+
+		resolvedFileContent, err = contentOfFile(resolvedFilePath)
+		if err != nil {
+			return false, err
+
+		} else if resolvedFileContent == "" {
+			return false, fmt.Errorf("Catfile.resolved is empty")
+		}
+		log.Debugf(resolvedFileName + " content: " + resolvedFileContent + "\n")
 	}
-	if resolvedFileContent == "" {
-		return false, nil
-	}
+
 	// ---
 
-	desiredCacheContent := fmt.Sprintf("--Swift version: %s --Swift version \n --%s: %s --%s", swiftVersion, resolvedFileName, resolvedFileContent, resolvedFileName)
-	return cacheFileContent == desiredCacheContent, nil
+	// Warn messages about the missing files
+
+	// Print the warning about the missing Cachefile only if the other required file (Cartfile.resolved) is available.
+	// If the Cartfile.resolved is not found, then we don't want to mislead the user with this warning.
+	if !cacheFileAvailable && resolvedFileAvailable {
+		fmt.Println()
+		log.Warnf("The " + cacheFileName + " is generated by the step. Probably cache not initialised yet (first cache push initialises the cache), nothing to worry about ;)")
+
+	}
+
+	if !resolvedFileAvailable {
+		fmt.Println()
+		log.Warnf("No "+resolvedFileName+" found at: %s", resolvedFilePath)
+		log.Warnf("Make sure it's committed into your repository!")
+		log.Warnf(resolvedFileName + " presence ensures that Bitrise will use exactly the same versions of dependencies as you in your local environment. ")
+		log.Warnf("The dependencies will not be cached until the " + resolvedFileName + " file is uploaded.")
```

Comment:
file is uploaded. -> file presents in the repository.

Reply:
Fixed in:
https://github.com/bitrise-steplib/steps-carthage/pull/54/commits/62b1a46e94bcfecee84de0dd713ca3fbb99f5415